testgen/src/main/groovy/com/vivy/openapi/GenerateApiDocsTest.groovy

```diff
@@ -0,0 +1,57 @@
+package com.vivy.openapi
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+
+class GenerateApiDocsTest extends DefaultTask {
+
+    @Input
+    String apiDocsTestOut;
+
+    @Input
+    @Optional
+    String apiDocsFileOut = "build/api-docs.json";
+
+    @Input
+    @Optional
+    String testPackage = "com.vivy.openapi";
+
+    @Input
+    String parentClass = null;
+
+    @TaskAction
+    def generate() {
+        def names = parentClass.split("\\.")
+        def parentClassSimpleName = names[names.length - 1];
+        new File(apiDocsTestOut).mkdirs()
+        new File(apiDocsTestOut, "OpenApiDocsTest.java").text =
+            """package $testPackage;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import $parentClass;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class OpenApiDocsTest extends $parentClassSimpleName {
+
+    @Autowired
+    protected TestRestTemplate restTemplate;
+
+    @Test
+    void generateAPIDocs() throws Exception {
+        var apiDocs = restTemplate.getForEntity("/v3/api-docs", String.class);
+        assertEquals(200, apiDocs.getStatusCodeValue());
+        Files.writeString(Path.of("$apiDocsFileOut"), apiDocs.getBody());
```

Comment:
please write to `build` dir results

Reply:
It is already build dir by default https://github.com/VivyTeam/openapi-boot-starter/pull/35/files/1b505f4bce882e77c3e7202cbcbd061fc1a5e1d7#diff-bd1f3c0a6d2a4da0a3b3c805ba97c80ae2952d2b7499569425e3ba92553d0d24R15

Reply:
can you make it typed then?

Reply:
Di not get what you mean

Reply:
It is no longer an input param.

Reply:
I meant that gradle can inject the project object where you can get a build dir as a File